hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -374,4 +416,212 @@ public static boolean areBothSpansFromDifferentService(Event event, Event parent
+                .filter(entry -> nullCheck(entry.getValue()))
+                .collect(
+                    Collectors.toUnmodifiableMap(
+                        entry -> entry.getKey().substring(GRPC_REQUEST_METADATA_PREFIX.length()),
+                        entry -> entry.getValue().getValue())));
+
+    // If the rpc system is grpc then read the keys with prefix "rpc.request.metadata"
+    if (isRpcSystemGrpc(attributes)) {
+      requestMetadataMap.putAll(
+          attributes.entrySet().stream()
+              .filter(entry -> entry.getKey().startsWith(RPC_REQUEST_METADATA_PREFIX))
+              .filter(entry -> nullCheck(entry.getValue()))
+              .collect(
+                  Collectors.toUnmodifiableMap(
+                      entry -> entry.getKey().substring(RPC_REQUEST_METADATA_PREFIX.length()),
+                      entry -> entry.getValue().getValue())));
+    }
+
+    return Collections.unmodifiableMap(requestMetadataMap);
+  }
+
+  public static Map<String, String> getGrpcResponseMetadata(Event event) {
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Collections.emptyMap();
+    }
+    Map<String, AttributeValue> attributes = event.getAttributes().getAttributeMap();
+    Map<String, String> responseMetadataMap = new HashMap<>();
+    // try to read keys with the prefix "grpc.response.metadata."
+    responseMetadataMap.putAll(
+        attributes.entrySet().stream()
+            .filter(entry -> entry.getKey().startsWith(GRPC_RESPONSE_METADATA_PREFIX))
+            .filter(entry -> nullCheck(entry.getValue()))
+            .collect(
+                Collectors.toUnmodifiableMap(
+                    entry -> entry.getKey().substring(GRPC_RESPONSE_METADATA_PREFIX.length()),
+                    entry -> entry.getValue().getValue())));
+
+    // If the rpc system is grpc then read the keys with prefix "rpc.response.metadata"
+    if (isRpcSystemGrpc(attributes)) {
+      responseMetadataMap.putAll(
+          attributes.entrySet().stream()
+              .filter(entry -> entry.getKey().startsWith(RPC_RESPONSE_METADATA_PREFIX))
+              .filter(entry -> nullCheck(entry.getValue()))
+              .collect(
+                  Collectors.toUnmodifiableMap(
+                      entry -> entry.getKey().substring(RPC_RESPONSE_METADATA_PREFIX.length()),
+                      entry -> entry.getValue().getValue())));
+    }
+
+    return Collections.unmodifiableMap(responseMetadataMap);
+  }
+
+  /**
+   * The request cookies are populated as `http.request.header.cookie` with value as
+   * cookie1=value1;cookie2=value2;cookie3=value3
+   *
+   * @return map of cookie key -> value
+   *     <ul>
+   *       <li>cookie1 -> value1
+   *       <li>cookie2 -> value2
+   *       <li>cookie3 -> value3
+   *     </ul>
+   */
+  public static Map<String, String> getRequestCookies(Event event) {
```

Comment:
Now I think, why not create 2 separate Util class in parallel to this: SpanHeaderUtils and SpanCookieUtils instead of fitting all in a single God class.